test/runtests.jl

```diff
@@ -141,6 +141,63 @@ end
     @test isapprox(coef(m2p_dep_pos_kw), coef(m2p))
 end
 
+# issue about can't converge within 30 iterations discussed in PR #314
+@testset "rankdeficient GLM" begin
+    # an example of rank deficiency caused by linearly dependent columns
+    num_rows = 100_000
+    dfrm = DataFrame()
+    dfrm[!, :x1] = randn(MersenneTwister(123), num_rows)
+    dfrm[!, :x2] = randn(MersenneTwister(456), num_rows)
+    dfrm[!, :x3]= 2*dfrm[!, :x1] + 3*dfrm[!, :x2]
+    dfrm[!, :y] = Int.(randn(MersenneTwister(9999999), num_rows) .> 0)
+    f1 = @eval(@formula(y ~ 1+x1+x2+x3))
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f1,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink())
+    @test_throws PosDefException fit(GeneralizedLinearModel,
+                                     f1,
+                                     dfrm,
+                                     Binomial(),
+                                     GLM.LogitLink();
+                                     allowrankdeficient=false)
+    m1 = fit(GeneralizedLinearModel,
+             f1,
+             dfrm,
+             Binomial(),
+             GLM.LogitLink();
+             allowrankdeficient=true)
+    @test isa(m1.model.pp.chol, CholeskyPivoted)
+    @test rank(m1.model.pp.chol) == 3
+    # Evaluated: 138626.46758072695 ≈ 138625.6633724341
+#     @test deviance(m1.model) ≈ 138625.6633724341
```

Comment:
@DilumAluthge I'm trying to rebase it, and it seems to be almost done except for these two deviance numbers. Could you tell me where did you get them?

Reply:
I reverted all my code back to commit `@e8a922dd`, and got the same result:
```julia
Test Summary: | Pass  Total
rankdeficient |   15     15
rankdeficient GLM: Test Failed at /home/xua/.julia/dev/GLM/test/runtests.jl:172
  Expression: deviance(m1.model) ≈ 138625.6633724341
   Evaluated: 138626.46758019557 ≈ 138625.6633724341
Stacktrace:
 [1] top-level scope at /home/xua/.julia/dev/GLM/test/runtests.jl:172
 [2] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1119
 [3] top-level scope at /home/xua/.julia/dev/GLM/test/runtests.jl:146
rankdeficient GLM: Test Failed at /home/xua/.julia/dev/GLM/test/runtests.jl:193
  Expression: deviance(m2.model) ≈ 138615.90834086522
   Evaluated: 138624.2610477953 ≈ 138615.90834086522
```
So would you check the benchmark data? Maybe it has been changed after the julia update

Reply:
It's been so long, I don't remember where I got those original numbers.

Reply:
I would remove those two tests.

Reply:
OK，then this PR could be ready to be merged @andreasnoack

Reply:
I'd rather uncomment these checks, but use the new value. That way if we break something in the future we'll notice it. Can you check these against e.g. R?

It would also be good to check the values of coefficients (here and below).